### PR TITLE
Patches for osx app bundle

### PR DESCRIPTION
**Pull request by ghost**
Now localization and ncurses build should be fine. @remyroy please help me test.

### CODE REVIEW

**Comment by remyroy** [comment]
libncurses is already provided with the base OS X. I just tested that with a clean version of OS X 10.7, 10.9 and 10.11 yesterday. It is the same version (5.4) across those OS as well. I don't think there is a need to bundle it.

libiconv is not a dependency when I build my app with `LOCALIZE=1`. Only libintl is. I believe I got my gettext library with brew. Different build procedures might require it but that would require more investigation.

To list your dependencies, just run `otool -L [executable]`. Here are the typical dependencies I get using this build command:

```
Remys-MacBook-Pro:Cataclysm-DDA remyroy$ make NATIVE=osx OSX_MIN=10.7 RELEASE=1 FRAMEWORK=1 CLANG=1 USE_HOME_DIR=1 CCACHE=1 LUA=1 LOCALIZE=1 TILES=1 SOUND=1 LANGUAGES=all -j4 dmgdist
```

```
Remys-MacBook-Pro:Cataclysm-DDA remyroy$ otool -L cataclysm-tiles 

cataclysm-tiles:

/usr/local/opt/gettext/lib/libintl.8.dylib (compatibility version 10.0.0, current version 10.1.0)

@rpath/SDL2_mixer.framework/Versions/A/SDL2_mixer (compatibility version 1.0.0, current version 1.1.0)

@rpath/Vorbis.framework/Versions/A/Vorbis (compatibility version 7.0.0, current version 7.5.0)

@rpath/Ogg.framework/Versions/A/Ogg (compatibility version 9.0.0, current version 9.1.0)

/usr/local/opt/lua/lib/liblua.5.2.dylib (compatibility version 5.2.0, current version 5.2.4)

/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)

@rpath/SDL2.framework/Versions/A/SDL2 (compatibility version 1.0.0, current version 5.0.0)

@rpath/SDL2_image.framework/Versions/A/SDL2_image (compatibility version 1.0.0, current version 1.1.0)

@rpath/SDL2_ttf.framework/Versions/A/SDL2_ttf (compatibility version 15.0.0, current version 15.0.0)

/System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa (compatibility version 1.0.0, current version 20.0.0)

/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.0.0)
```

**Comment by remyroy** [comment]
libiconv is a dependency of libintl.8.dylib on my system but it uses the libiconv provided by the OS and it works fine across 10.7, 10.9 and 10.11.  I don't think there is a need to bundle it.

**Comment by jakwings** [comment]
Don't know if an older ncursesw has some bugs about text layout, I need to check.

Could you tell me whether 10.7 and 10.9 have libiconv?

**Comment by remyroy** [comment]
10.7 and 10.9 both have libiconv

File: data/osx/Cataclysm.sh [hunk 1]
```diff
@@ -4,6 +4,7 @@ cd "${PWD}/../Resources/"
 
 COMMAND='
 OSREV=`uname -r | cut -d. -f1`;
+export TERMINFO=/usr/share/terminfo;
```
Comment (remyroy) [inline data/osx/Cataclysm.sh]: There is no need to export `TERMINFO=/usr/share/terminfo;`. According to the apple [terminfo manual](https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man5/terminfo.5.html):

> If neither TERMINFO nor TERMINFO_DIRS is set, the last place tried will be the system terminfo directory, directorytory, /usr/share/terminfo.

Comment (jakwings) [inline data/osx/Cataclysm.sh]: I'm not sure, even before the apple script was introduced, I had already met this problem: `Error opening terminal: xterm-256color`.

Comment (remyroy) [inline data/osx/Cataclysm.sh]: Do you have a custom `TERMINFO` or a `TERMINFO_DIRS` in your environment variables? Type `printenv` in a Terminal to find out.

Comment (jakwings) [inline data/osx/Cataclysm.sh]: Nothing.

But after I revert all three commits, uninstall libiconv and ncurses, it seems completely ok now. So I guess the problem is caused by libncursesw from homebrew.

**Comment by jakwings** [comment]
Thanks for helping!  Let's closing this for now. It seems tricky to use another libraries the system already has. The MacPorts build is facing the same problem as gettext depends on both libiconv and ncurses (though I don't why it is by default). I'll investigate.

**Pull request closed without merging**